game-core/src/main/java/games/strategy/engine/lobby/client/login/ForgotPasswordPanel.java

```diff
@@ -0,0 +1,151 @@
+
+  private ForgotPasswordPanel() {
+    layoutComponents();
+    setupListeners();
+  }
+
+  /**
+   * Creates a new instance of the {@code CreateUpdateAccountPanel} class used to request a
+   * temporary password.
+   *
+   * @return A new {@code CreateUpdateAccountPanel}.
+   */
+  static ForgotPasswordPanel newForgotPasswordPanel() {
+    return new ForgotPasswordPanel();
+  }
+
+  private void layoutComponents() {
+    setLayout(new BorderLayout());
+    final JLabel label = new JLabel(new ImageIcon(Util.getBanner(title)));
+    add(label, BorderLayout.NORTH);
+
+    final JPanel main = new JPanel();
+    add(main, BorderLayout.CENTER);
+    main.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
+    main.setLayout(new GridBagLayout());
+    main.add(
+        new JLabel("Username:"),
+        new GridBagConstraints(
+            0,
+            0,
+            1,
+            1,
+            0.0,
+            0.0,
+            GridBagConstraints.WEST,
+            GridBagConstraints.NONE,
+            new Insets(0, 0, 0, 0),
+            0,
+            0));
+    main.add(
+        userNameField,
+        new GridBagConstraints(
+            1,
+            0,
+            1,
+            1,
+            1.0,
+            0.0,
+            GridBagConstraints.WEST,
+            GridBagConstraints.HORIZONTAL,
+            new Insets(0, 5, 0, 0),
+            0,
+            0));
+
+    final JPanel buttons = new JPanel();
+    add(buttons, BorderLayout.SOUTH);
+    buttons.setBorder(BorderFactory.createEmptyBorder(10, 5, 10, 5));
+    buttons.setLayout(new FlowLayout(FlowLayout.RIGHT, 5, 0));
+    buttons.add(okButton);
+    buttons.add(cancelButton);
+  }
+
+  private void setupListeners() {
+    cancelButton.addActionListener(e -> close());
+    okButton.addActionListener(e -> okPressed());
+
+    SwingComponents.addEnterKeyListener(this, this::okPressed);
+  }
+
+  private void close() {
+    if (dialog != null) {
+      dialog.setVisible(false);
+    }
+  }
+
+  private void okPressed() {
+    if (!PlayerNameValidation.isValid(userNameField.getText())) {
+      JOptionPane.showMessageDialog(
+          this,
+          PlayerNameValidation.validate(userNameField.getText()),
+          "Invalid name",
+          JOptionPane.ERROR_MESSAGE);
+      return;
+    }
+
+    returnValue = ReturnValue.OK;
+    close();
+  }
+
+  /**
+   * Shows this panel in a modal dialog.
+   *
+   * @param parent The dialog parent window.
+   * @return {@link ReturnValue#OK} if the user clicks 'okay' button, otherwise {@link
+   *     ReturnValue#CANCEL}.
+   */
+  ReturnValue show(final Window parent) {
+    dialog = new JDialog(JOptionPane.getFrameForComponent(parent), title, true);
+    dialog.getContentPane().add(this);
+    SwingComponents.addEscapeKeyListener(dialog, this::close);
+    dialog.pack();
+    dialog.setLocationRelativeTo(parent);
+    dialog.setVisible(true);
+    dialog.dispose();
+    dialog = null;
+    return returnValue;
```

Comment:
I'm a little confused I'd have assumed JOptionPane in combination with your swing helper library would reduce the amount of code?

Reply:
It very likely would have. Compare the the two new panel classes to: (CreateUpdateAccountPanel)[https://github.com/triplea-game/triplea/blob/master/game-core/src/main/java/games/strategy/engine/lobby/client/login/CreateUpdateAccountPanel.java] and you'll see a lot of similarity. So in essence we're looking at glorified copy/pastes. It's not consistent if these panels deviate too much in their implementation.